index.d.ts

```diff
@@ -94,6 +95,17 @@ export interface FetchOptions {
     retryDelayOptions?: RetryDelayOption
 }
 
+//Plugins
+export interface ContentstackPlugin {
+    onRequest?(stack: Stack, request: ContentstackRequest): ContentstackRequest;
+    onResponse?(stack: Stack, request: ContentstackRequest, response: any, data: any): any;
+}
+
+export interface ContentstackRequest {
+    url: string;
+    headers: object;
```

Comment:
change headers to `option` as per implementation here: https://github.com/contentstack/contentstack-javascript/blob/33df49f790ecb277d3d260a1ab9fa33731019397/src/core/lib/request.js#L101